main.js

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node
+
+const env = require('./.env');
+const DB = process.argv[2] || env.from.DB;
+const USER = process.argv[3] || env.from.USER;
+const PASS = process.argv[4] || env.from.PASS;
+const HOST = process.argv[5] || env.from.HOST;
+
+if (!DB || !USER || !PASS || !HOST) {
+  console.error('Some configuration is missing, please set the .env file with the configuration or pass it through parameters');
+  process.exit(1);
+}
+
+const fs = require('fs');
+const path = require('path');
+const mysql = require('./config/mysql.js');
+const dateFormat = require('./utils/dateFormat.js');
+const migrations = require('./utils/migrations.js');
+const writeLog = require('./utils/writeLog.js');
+const chalk = require('chalk');
+const start = Date.now();
+
+const createMigrationFile = (file_data, table_name, db_name, isView = false) => {
+  return new Promise((resolve, reject) => {
+    // Creating migrations' directory
+    const pathMigrations = './migrations';
+    fs.mkdir(path.resolve(__dirname, pathMigrations), { recursive: true }, (err) => {
+      if (err) {
+        writeLog(err);
+      }
+      const pathToWrite = !isView ? `${pathMigrations}/${db_name}_tables` : `${pathMigrations}/${db_name}_views`;
+      fs.mkdir(path.resolve(__dirname, pathToWrite), { recursive: true }, (err) => {
+        if (err) {
+          writeLog(err);
+        }
+        // Writing file in its respective directory
+        const filename = `${dateFormat(new Date())}_create_${table_name.toLowerCase()}.js`;
+        const stream = fs.createWriteStream(path.resolve(__dirname, `${pathToWrite}/${filename}`));
+        stream.write(file_data);
+        // Returning stream promise
+        stream.on('error', reject);
+        stream.on('open', () => {
```

Comment:
The open event is probably not when we should give the user this feedback. It should be the end event.